Sources/muterCore/MutationTesting/MutationTestingIODelegate.swift

```diff
@@ -23,16 +23,16 @@ struct MutationTestingDelegate: MutationTestingIODelegate {
                       savingResultsIntoFileNamed fileName: String) -> (outcome: TestSuiteOutcome, testLog: String) {
         do {
             let (testProcessFileHandle, testLogUrl) = try fileHandle(for: fileName)
+            defer { testProcessFileHandle.closeFile() }
```

Comment:
Is it the same as we had before? It seems now that the file handle is closed after we attempt to read it.

Reply:
defer happens right before the scope closes. It should be the same behavior as before, but now it'll still happen if we add an early return. I was originally going to use a `guard` in an earlier version of this PR. I ended up not doing it, but I still like using `defer` for `open`/`close` patterns.